### PR TITLE
Setting extra-vars as their own var

### PR DESCRIPTION
**Pull request by jpodivin**

### CODE REVIEW

File: examples/openstack-ansibleee-play-runner-extra-vars.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            path: ssh_key
+      mounts:
+      - name: inventory
+        mountPath: /runner/inventory/inventory.yaml
+        subPath: inventory.yaml
+      - name: sshkey
+        mountPath: /runner/env/ssh_key
+        subPath: ssh_key
+  extraVars:
+    secretmsg: "Hello Mars!"
```
Comment (fultonj) [inline examples/openstack-ansibleee-play-runner-extra-vars.yaml:34]: This is a more convenient way to pass `extraVars`. Thank you Jiri.

**Review by slagle — COMMENTED** [review]
I'm ok with adding this. We don't yet have a specific use case for how this would be used from dataplane-operator, however if we add an AnsibleVars concept to OpenStackDataPlaneDeployment, I could see that being implemented by using the ExtraVars added here.

We do need to keep in mind that openstack-ansibleee-operator isn't meant to be a general purpose ansible operator, so we don't need to go enabling every ansible-runner interface. It's reallly just meant for the specific EDPM use cases and consumed by dataplane-operator.

File: tests/functional/ansibleee_controller_test.go [hunk 2]
```diff
@@ -309,5 +311,28 @@ var _ = Describe("Ansibleee controller", func() {
 			})
 		})
 
+		Context("with extra vars", func() {
+			extraVars := map[string]json.RawMessage{
+				"foo":  json.RawMessage("bar"),
+				"fizz": json.RawMessage("buzz"),
```
[the remark this answers is not in the document]
Comment (jpodivin) [reply tests/functional/ansibleee_controller_test.go:317]: That doesn't seem to work either. Instead I'm getting

```
json: error calling MarshalJSON for type json.RawMessage: invalid character 'b' looking for beginning of value
```
I'll post new version.

File: examples/openstack-ansibleee-play-runner-extra-vars.yaml (hunk 1 above)
Comment (jpodivin) [reply examples/openstack-ansibleee-play-runner-extra-vars.yaml:34]: You are welcome :)

**Comment by rebtoor** [comment]
/test ansibleee-operator-build-deploy-kuttl

```
Go compliance shim [305] [][]: invoking real go binary
go: k8s.io/api@v0.26.9: Get "https://proxy.golang.org/k8s.io/api/@v/v0.26.9.mod": EOF
Go compliance shim [305] [][]: Exited with: 1
```

**Comment by jlarriba** [comment]
> I'm ok with adding this. We don't yet have a specific use case for how this would be used from dataplane-operator, however if we add an AnsibleVars concept to OpenStackDataPlaneDeployment, I could see that being implemented by using the ExtraVars added here.
> 
> We do need to keep in mind that openstack-ansibleee-operator isn't meant to be a general purpose ansible operator, so we don't need to go enabling every ansible-runner interface. It's reallly just meant for the specific EDPM use cases and consumed by dataplane-operator.

I think there is actually a use case: to pass the names of the images specified directly in the operator's CR, without any need of having to modify the ansible vars in edpm_ansible.

**Pull request merged**